internal/api/v1/organizations.go

```diff
@@ -161,3 +155,53 @@ func (oc OrganizationsController) Delete(w http.ResponseWriter, r *http.Request)
 
 	return nil
 }
+
+func deleteApps(cluster *kubernetes.Cluster, gitea *gitea.Client, org string) error {
+	apps, err := application.List(cluster, org)
+	if err != nil {
+		return err
+	}
+
+	const maxConcurrent = 100
+	errChan := make(chan error, maxConcurrent)
+	success := make(chan struct{})
+	stop := make(chan struct{})
+
+	go func() {
+		var wg sync.WaitGroup
+		// create a buffer, so there is no more than 'maxConcurrent' goroutines running at the same time
+		buffer := make(chan struct{}, maxConcurrent)
+	loop:
+		for _, app := range apps {
+			buffer <- struct{}{}
+
+			go func(app application.Application) {
+				defer wg.Done()
+				wg.Add(1)
```

Comment:
It could happen that `wg.Wait()` is executed before any of the `wg.Add(1)` calls have been made. It's better to call `wg.Add(1)` right before the go routing call to ensure it always executes before `wg.Wait()`.

Reply:
yes I agree. Done in the next commit